plugin.py

```diff
@@ -43,6 +43,20 @@ def on_settings_read(cls, settings: sublime.Settings) -> bool:
 
         return False
 
+    def on_ready(self, api) -> None:
+        api.on_notification("pyright/beginProgress", self.handle_begin_progress)
+        api.on_notification("pyright/endProgress", self.handle_end_progress)
+        api.on_notification("pyright/reportProgress", self.handle_report_progress)
+
+    def handle_begin_progress(self, params) -> None:
+        sublime.status_message("{}: Progress begins".format(self.package_name))
+
+    def handle_end_progress(self, params) -> None:
+        sublime.status_message("{}: Progress ends".format(self.package_name))
```

Comment:
More like `Progress ended`. Also ideally we would just clear the previous message instead of posting this one which will pointlessly linger for a few seconds.

Reply:
Probably. Depends on whether the user wants to know it ended. If the last sub-progress takes a long time, it will auto disappear before it actually ends.

Reply:
Are you saying that the server will send it before the actual "task" ends? I would think that it sends it **after** the task ends so that the UI can clear the progress indicator.

But anyway, having this message stay for few seconds is not necessarily an issue.

Reply:
> Are you saying that the server will send it before the actual "task" ends? I would think that it sends it **after** the task ends so that the UI can clear the progress indicator.

To be clear, take the screen from this PR for an example.

```
LSP-xxx: 3 files to analyze
LSP-xxx: 2 files to analyze
LSP-xxx: 1 files to analyze <--------- if somehow it takes a long time to analyze the file, it's so long that the status message auto disappear and we don't show the next "Progress ended", then the user may not know whether it ends or not.
LSP-xxx: Progress ended
```

Reply:
True but then it feels like it should use something like a `sublime_lib.ActivityIndicator` indicator so that the message doesn't disappear when there is task in progress...

Reply:
Agree.

Reply:
There is `Session.set_window_status_async` and `Session.erase_window_status_async` for exactly this purpose. (although it's not yet released, and also ST4-only).

Reply:
Will expose that in the lsp_utils refactor using a common API then.

Reply:
Sounds like we have to decide whether we want to maintain 2 branches of this plugin?